vol_dataset_test.c

```diff
@@ -3989,11 +3989,210 @@ test_dataset_property_lists(void)
+        TEST_ERROR;
+
+    for (size_t i = 0; i < DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT; i++)
+        for (size_t j = 0; j < DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT; j++)
+            ((int *)write_buf)[i * DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT + j] = j;
+
+    PASSED();
+
+    BEGIN_MULTIPART
+    {
+        PART_BEGIN(H5Dget_storage_size_contiguous)
+        {
+            TESTING_2("H5Dget_storage_size on contiguous dataset");
+
+            if (H5Dwrite(dset_id_contiguous, type_id, H5S_ALL, H5S_ALL, H5P_DEFAULT,
+                         (const void *)write_buf) < 0) {
+                H5_FAILED();
+                HDprintf("    couldn't write to contiguous dataset\n");
+                PART_ERROR(H5Dget_storage_size_contiguous);
+            }
+
+            if (H5Dget_storage_size(dset_id_contiguous) <= 0) {
+                H5_FAILED();
+                HDprintf("    got incorrect storage size for contiguous dataset\n");
+                PART_ERROR(H5Dget_storage_size_contiguous);
+            }
+
+            PASSED();
+        }
+        PART_END(H5Dget_storage_size_contiguous);
+
+        PART_BEGIN(H5Dget_storage_size_chunked)
+        {
+            TESTING_2("H5Dget_storage_size of chunked dataset");
+
+            if (H5Dwrite(dset_id_chunked, type_id, H5S_ALL, H5S_ALL, H5P_DEFAULT, (const void *)write_buf) <
+                0) {
+                H5_FAILED();
+                HDprintf("    couldn't write to chunked dataset\n");
+                PART_ERROR(H5Dget_storage_size_chunked);
+            }
+
+            if (H5Dget_storage_size(dset_id_chunked) <= 0) {
+                H5_FAILED();
+                HDprintf("    got incorrect storage size for chunked dataset\n");
+                PART_ERROR(H5Dget_storage_size_chunked);
+            }
+
+            PASSED();
+        }
+        PART_END(H5Dget_storage_size_chunked);
+
+        PART_BEGIN(H5Dget_storage_size_filtered)
+        {
+            TESTING_2("H5Dget_storage_size of filtered dataset");
+
+            if (!(vol_cap_flags_g & H5VL_CAP_FLAG_FILTERS)) {
+                SKIPPED();
+                HDprintf("    dataset filters are not supported by this VOL connector\n");
+                PART_EMPTY(H5Dget_storage_size_filtered);
+            }
+
+            if (H5Dwrite(dset_id_filtered, type_id, H5S_ALL, H5S_ALL, H5P_DEFAULT, (const void *)write_buf) <
+                0) {
+                H5_FAILED();
+                HDprintf("    couldn't write to filtered dataset\n");
+                PART_ERROR(H5Dget_storage_size_filtered);
+            }
+
+            if (H5Dget_storage_size(dset_id_filtered) <= 0) {
+                H5_FAILED();
+                HDprintf("    got incorrect storage size for filtered dataset\n");
+                PART_ERROR(H5Dget_storage_size_filtered);
+            }
+
+            PASSED();
+        }
+        PART_END(H5Dget_storage_size_filtered);
+    }
+    END_MULTIPART;
+
+    free(write_buf);
```

Comment:
Should set write_buf to NULL so if a close below fails we won't double free in the error: cleanup section.